cloudformation/cf-resources.yaml

```diff
@@ -0,0 +1,624 @@
+      Specifies the number of instances to launch and register to the cluster.
+      Defaults to 1.
+    Default: '1'
+  IamRoleInstanceProfile:
+    Type: String
+    Description: >
+      Specifies the Name or the Amazon Resource Name (ARN) of the instance
+      profile associated with the IAM role for the instance
+    Default: ScorekeepInstanceProfile
+  SecurityIngressFromPort:
+    Type: Number
+    Description: >
+      Optional - Specifies the Start of Security Group port to open on
+      ECS instances - defaults to port 0
+    Default: '80'
+  SecurityIngressToPort:
+    Type: Number
+    Description: >
+      Optional - Specifies the End of Security Group port to open on ECS
+      instances - defaults to port 65535
+    Default: '80'
+  SecurityIngressCidrIp:
+    Type: String
+    Description: >
+      Optional - Specifies the CIDR/IP range for Security Ports - defaults
+      to 0.0.0.0/0
+    Default: 0.0.0.0/0
+  EcsEndpoint:
+    Type: String
+    Description: >
+      Optional - Specifies the ECS Endpoint for the ECS Agent to connect to
+    Default: ''
+  RootEbsVolumeSize:
+    Type: Number
+    Description: >
+      Optional - Specifies the Size in GBs of the root EBS volume
+    Default: 30
+  EbsVolumeSize:
+    Type: Number
+    Description: >
+      Optional - Specifies the Size in GBs of the data storage EBS volume used by the Docker in the AL1 ECS-optimized AMI
+    Default: 22
+  EbsVolumeType:
+    Type: String
+    Description: Optional - Specifies the Type of (Amazon EBS) volume
+    Default: 'gp2'
+    AllowedValues:
+      - ''
+      - standard
+      - io1
+      - gp2
+      - sc1
+      - st1
+    ConstraintDescription: Must be a valid EC2 volume type.
+  RootDeviceName:
+    Type: String
+    Description: Optional - Specifies the device mapping for the root EBS volume.
+    Default: /dev/xvda
+  DeviceName:
+    Type: String
+    Description: Optional - Specifies the device mapping for the EBS volume used for data storage. Only applicable to AL1.
+    Default: /dev/xvdcz
+  UseSpot:
+    Type: String
+    Default: 'false'
+  IamSpotFleetRoleArn:
+    Type: String
+    Default: ''
+  SpotPrice:
+    Type: String
+    Default: ''
+  SpotAllocationStrategy:
+    Type: String
+    Default: 'diversified'
+    AllowedValues:
+      - 'lowestPrice'
+      - 'diversified'
+  UserData:
+    Type: String
+    Default: |
+      #!/bin/bash
+      echo ECS_CLUSTER=scorekeep-cluster >> /etc/ecs/ecs.config;echo ECS_BACKEND_HOST= >> /etc/ecs/ecs.config;
+  IsWindows:
+    Type: String
+    Default: 'false'
+  ConfigureRootVolume:
+    Type: String
+    Description: Optional - Specifies if there should be customization of the root volume
+    Default: 'true'
+  ConfigureDataVolume:
+    Type: String
+    Description: Optional - Specifies if there should be customization of the data volume
+    Default: 'false'
+  AutoAssignPublicIp:
+    Type: String
+    Default: 'INHERIT'
+Conditions:
+  UseT2MicroInstance:
```

Comment:
UseT2MicroInstance condition needs review.
Here it is checked that the AWS region does not match any of the listed regions. If there is a match, the condition is false (T3 micro instance will be used instead of T2).

Reply:
Oof, this is a painful piece of config because it may require upkeep.
There is no way to get it programmatically as far as I can tell.
Can we do containsvalue instead? It will be easier to read.

https://stackoverflow.com/questions/61727244/how-to-use-an-array-in-a-condition-for-azure-devops-yaml

Reply:
I was unable to find a Cloudformation equivalent of containsValue.
An alternative was to use Cloudformation "Mappings", however there is no default value in mappings, so in this case every region would need to be mapped to either t2.micro or t3.micro, which would be more difficult to maintain.
Currently, this conditional is the best workaround I could find.

Reply:
Sounds good.